### PR TITLE
Disable jwt issuedAt verification

### LINKED ISSUE #15047: 8.3.x release benchmark - Authorization data unavailable: The Token can't be used before xxx
**Describe the bug**

This error happened on Sat 03/11:

```
"StreamProcessor-1 failed, marking it as dead: StreamProcessor-1{status=DEAD, issue=HealthIssue[message=null, throwable=io.camunda.zeebe.util.exception.UnrecoverableException: Authorization data unavailable: The Token can't be used before 2023-11-03T14:46:52Z., cause=null]}"
```

You can find the logs [here](https://drive.google.com/drive/folders/1wJqLwgla0px44joF49Ve0bnW0WzZvtQC)

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Disable the `issuedAt` verification for Zeebe JWT tokens. 

Zeebe only uses unsigned JWT tokens for transporting data internally, so validating an `issuedAt` claim is not necessary. 

It actually may cause the `stream processor` to fail if the Gateway and Broker clocks are (even slightly) out of sync. If the Broke clock is behing the Gateway clock, an error showing `Authorization data unavailable: The Token can't be used before xxx` is thrown.

Note: I decided to only disable the `issuedAt` verification, but still verify the JWT token, since it's useful to detect if there are cases where the `AUTHORIZED_TENANTS` claim is not included.

## Related issues

closes #15047

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: auth/src/main/java/io/camunda/zeebe/auth/impl/JwtAuthorizationDecoder.java [hunk 1]
```diff
@@ -122,4 +111,25 @@ public JwtAuthorizationDecoder withJwtToken(final String token) {
...
+      verificationBuilder.withClaimPresence(claim);
+    }
+    final JWTVerifier jwtVerification = verificationBuilder.build();
+
+    try {
+      return jwtVerification.verify(jwtToken);
+    } catch (final JWTVerificationException | NullPointerException ex) {
+      LOGGER.error("Authorization data unavailable: {}", ex.getMessage());
+      throw new UnrecoverableException("Authorization data unavailable: " + ex.getMessage(), ex);
+    }
```
Comment (korthout) [inline auth/src/main/java/io/camunda/zeebe/auth/impl/JwtAuthorizationDecoder.java:133]: 🔧 The try-catch is now duplicated between here and lines 76-81. I suggest inlining this method again, as all the builder does is validation (pretty much).

**Review by korthout — APPROVED** [review]
Thanks @koevskinikola ❤️ 

👍 Great fix with nice and simple test!

❓ Do I understand correctly that this also resolves the issue when the leeway is slightly skewed?

LGTM 👍 

🔧 I have just one suggestion

**Comment by koevskinikola** [comment]
bors merge

**Comment by korthout** [comment]
Thanks @koevskinikola for switching the label :) 

/backport

**Pull request merged**